packages/daemon/src/host.js

```diff
@@ -514,12 +523,39 @@ export const makeHostMaker = ({
       return cancelValue(formulaIdentifier, reason);
     };
 
+    /** @type {import('./types.js').EndoHost['gateway']} */
+    const gateway = async () => {
+      // TODO: we should only materialize local objects this way
+      return Far('Gateway', { provide: provideValueForFormulaIdentifier });
+    };
+
+    /** @type {import('./types.js').EndoHost['addPeerInfo']} */
+    const addPeerInfo = async peerInfo => {
+      const endoBootstrap =
+        /** @type {import('./types.js').FarEndoBootstrap} */ (
+          await lookup('ENDO')
+        );
+      await E(endoBootstrap).addPeerInfo(peerInfo);
+    };
+
+    /** @type {import('./types.js').EndoHost['getPeerInfo']} */
+    const getPeerInfo = async () => {
+      const addresses = await getAllNetworkAddresses(
+        networksDirectoryFormulaIdentifier,
+      );
+      const peerInfo = {
+        location: ownLocation,
```

Comment:
Location is going to become quickly ambiguous since we will have locations of varying specificity:

- `endo://node` node location
- `endo://node/value` value location
- `endo://node/value.type` typed value location
- `endo://node/value.type?address=&address=` value location (née offline capability) with connection hints
- `endo://node/value.blob-reader-base64/application/javascript?address=&address=` MIME-typed downloadable blob location over base64 until CapTP maybe someday can do byte streams

This can perhaps just be `peer` implying “peer identifier” or “peer public key” someday